internal/eval/store_eval.go

```diff
@@ -295,32 +264,27 @@ func evalSET(args []string, store *dstore.Store) *EvalResponse {
 
 			// if key does not exist, return RESP encoded nil
 			if obj == nil {
-				return &EvalResponse{
-					Result: clientio.NIL,
-					Error:  nil,
-				}
+				return makeEvalResult(clientio.NIL)
 			}
 		case NX:
 			obj := store.Get(key)
 			if obj != nil {
-				return &EvalResponse{
-					Result: clientio.NIL,
-					Error:  nil,
-				}
+				return makeEvalResult(clientio.NIL)
 			}
 		case KeepTTL:
 			if state != Uninitialized {
-				return &EvalResponse{
-					Result: nil,
-					Error:  diceerrors.ErrSyntax,
-				}
+				return makeEvalError(diceerrors.ErrSyntax)
 			}
 			keepttl = true
-		default:
-			return &EvalResponse{
-				Result: nil,
-				Error:  diceerrors.ErrSyntax,
+		case GET:
+			getVal = true
+			oldVal := evalGET([]string{key}, store)
+			if oldVal.Error != nil {
+				return makeEvalError(diceerrors.ErrWrongTypeOperation)
 			}
+			getOldVal = oldVal.Result
```

Comment:
can we instead make getOldVal a pointer to Result?
that way we will not need the getVal variable on line 305

Reply:
Can you please share this as a code suggestion? that way I would be able to accurately test it in local.